test/milton/MiltonShouldCalculate.t.sol

```diff
@@ -0,0 +1,737 @@
+            TestConstants.ZERO_INT
+        );
+        _usdtMockedToken = getTokenUsdt();
+        _usdcMockedToken = getTokenUsdc();
+        _daiMockedToken = getTokenDai();
+        _ipTokenUsdt = getIpTokenUsdt(address(_usdtMockedToken));
+        _ipTokenUsdc = getIpTokenUsdc(address(_usdcMockedToken));
+        _ipTokenDai = getIpTokenDai(address(_daiMockedToken));
+        _admin = address(this);
+        _userOne = _getUserAddress(1);
+        _userTwo = _getUserAddress(2);
+        _userThree = _getUserAddress(3);
+        _liquidityProvider = _getUserAddress(4);
+        _users = usersToArray(_admin, _userOne, _userTwo, _userThree, _liquidityProvider);
+    }
+
+    function testShouldCalculateIncomeFeeFivePercentWhenReceiveFixedAndMiltonLosesAndUserEarnsAndDepositIsGreaterThanDifferenceBetweenLegsAfterMaturity(
+    ) public {
+        // given
+        _miltonSpreadModel.setCalculateQuoteReceiveFixed(TestConstants.PERCENTAGE_10_18DEC);
+        ItfIporOracle iporOracle =
+            getIporOracleAsset(_userOne, address(_daiMockedToken), TestConstants.TC_120_EMA_18DEC_64UINT);
+        MockCase1Stanley stanleyDai = getMockCase1Stanley(address(_daiMockedToken));
+        MiltonStorage miltonStorageDai = getMiltonStorage();
+        MockCase2MiltonDai mockCase2MiltonDai = getMockCase2MiltonDai(
+            address(_daiMockedToken),
+            address(iporOracle),
+            address(miltonStorageDai),
+            address(_miltonSpreadModel),
+            address(stanleyDai)
+        );
+        MockCase0JosephDai mockCase0JosephDai = getMockCase0JosephDai(
+            address(_daiMockedToken),
+            address(_ipTokenDai),
+            address(mockCase2MiltonDai),
+            address(miltonStorageDai),
+            address(stanleyDai)
+        );
+        prepareApproveForUsersDai(_users, _daiMockedToken, address(mockCase0JosephDai), address(mockCase2MiltonDai));
+        prepareMilton(mockCase2MiltonDai, address(mockCase0JosephDai), address(stanleyDai));
+        prepareJoseph(mockCase0JosephDai);
+        prepareIpToken(_ipTokenDai, address(mockCase0JosephDai));
+        vm.prank(_liquidityProvider);
+        mockCase0JosephDai.itfProvideLiquidity(TestConstants.TC_LP_BALANCE_BEFORE_CLOSE_18DEC, block.timestamp);
+        openSwapReceiveFixed(
+            _userTwo,
+            block.timestamp,
+            TestConstants.TC_TOTAL_AMOUNT_10_000_18DEC,
+            TestConstants.PERCENTAGE_10_18DEC,
+            TestConstants.LEVERAGE_18DEC,
+            mockCase2MiltonDai
+        );
+        vm.prank(_userOne);
+        iporOracle.itfUpdateIndex(address(_daiMockedToken), TestConstants.PERCENTAGE_5_18DEC, block.timestamp);
+        ExpectedMiltonBalances memory expectedBalances;
+        expectedBalances.expectedPayoffAbs = 682671910755540429746;
+        expectedBalances.expectedIncomeFeeValue = 34133595537777021487;
+        int256 openerUserLost = TestConstants.TC_OPENING_FEE_18DEC_INT
```

Comment:
Please do one think in this file - simple refactor (which makes that it will be a little different in old test but for me acceptable):
Extract code where you create expected balance to separated method for example "createExpectedBalance(expectedPayoff, expectedIncomeFeeValue)"

Calculation when Milton earn or loose will be the same only what you have to do is not to use absolute value, then formulas in new method "createExpectedBalance" will be the same for all tests.

When you create this method then use it in every test in this file.